providers/os/id/awsebs/awsebs.go

```diff
@@ -0,0 +1,276 @@
+	case m.platform.IsFamily(inventory.FAMILY_WINDOWS):
+		return m.windowsMetadata()
+	default:
+		return nil, errors.New("your platform is not supported by aws metadata identifier resource")
+	}
+}
+
+func (m *ebsMetadata) metadataFromEC2Instance(instance *ec2types.Instance) (any, error) {
+	mdata := map[string]any{}
+	if privateDNS := convert.ToValue(instance.PrivateDnsName); privateDNS != "" {
+		mdata["hostname"] = privateDNS
+	}
+	if publicDNS := convert.ToValue(instance.PublicDnsName); publicDNS != "" {
+		mdata["public-hostname"] = publicDNS
+	}
+	if vpcID := convert.ToValue(instance.VpcId); vpcID != "" {
+		mdata["vpc-id"] = vpcID
+	}
+
+	macs := map[string]*macDetails{}
+	for _, nInterface := range instance.NetworkInterfaces {
+		mac := convert.ToValue(nInterface.MacAddress)
+		if mac == "" {
+			continue
+		}
+
+		md, exist := macs[mac]
+		if !exist {
+			md = &macDetails{
+				MAC:        mac,
+				LocalIPv4s: convert.ToValue(nInterface.PrivateIpAddress),
+			}
+			macs[mac] = md
+		}
+
+		if nInterface.Association != nil {
+			// We may have found a public ip
+			md.PublicIPv4s = convert.ToValue(nInterface.Association.PublicIp)
+		}
+	}
+
+	mdata["network"] = map[string]any{
+		"interfaces": map[string]any{
+			"macs": macs,
+		},
+	}
+	return mdata, nil
+}
+
+func (m *ebsMetadata) Identify() (Identity, error) {
+	log.Debug().Msg("getting ebs device identity")
+	identity := Identity{}
+
+	guid, err := machineid.MachineId(m.conn, m.platform)
+	if err != nil {
+		return identity, errors.Wrap(err, "unable to identify platform metadata")
+	}
+
+	identity.InstanceMachineID = guid
+
+	// if we get execute by our serverless offering, we will have an injected platform-id
+	if platformID, instanceID, ok := m.extractInjectedPlatformID(); ok {
+		identity.InstanceID = instanceID.Id
+		identity.PlatformIDs = []string{
+			"//platformid.api.mondoo.app/runtime/aws/accounts/" + instanceID.Account,
+			platformID,
+		}
+	} else if id, ok := m.getInstanceID(); ok {
+		// if we couldn't detect the injected platform-id, try to get information from the volume itself
+		identity.InstanceID = id
+		identity.PlatformIDs = []string{
+			"//platformid.api.mondoo.app/machineid/" + guid,
+			"//platformid.api.mondoo.app/aws/ebs/instances/" + id,
+		}
+	}
+
+	return identity, nil
+}
+
+var (
+	instanceIdRE = regexp.MustCompile(`i-[0-9a-f]{17}`)
+	regionRE     = regexp.MustCompile(`[a-z]{2}-[a-z]+-\d`)
+)
+
+func (m *ebsMetadata) extractInjectedPlatformID() (string, *awsec2.MondooInstanceId, bool) {
+	if asset := m.conn.Asset(); asset != nil {
+		connections := asset.GetConnections()
+		index := slices.IndexFunc(connections, func(c *inventory.Config) bool {
+			if c != nil && c.Type == shared.Type_Device.String() {
+				return true
+			}
+			return false
+		})
+		// @afiune we can't use `device.PlatformIdInject` because of a cyclic dep
+		// TODO move it to a shared package
+		platformInjected, ok := connections[index].Options["inject-platform-ids"]
```

Comment:
connections[index] will blow up here if it's -1 (not found)

Reply:
also it feels a bit weird to have the flag reading here as that's connection specific

i think we want to move away from this and just inject asset.PlatformIds directly instead of this option

Reply:
@preslavgerchev I'd love to hear you idea about injecting the platform IDs directly, I'll ping you to chat about it.

Reply:
Preslav and I synced and we decided to do this cleanup in a different change. That is, remove the `inject-platform-ids` from both, `cnquery` and `serverless` projects.